pygmt/src/subplot.py

```diff
@@ -0,0 +1,232 @@
+"""
+subplot - Manage modern mode figure subplot configuration and selection.
+"""
+import contextlib
+
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    build_arg_string,
+    fmt_docstring,
+    is_nonstr_iter,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@contextlib.contextmanager
+@use_alias(
+    Ff="figsize",
+    Fs="subsize",
+    A="autolabel",
+    B="frame",
+    C="clearance",
+    J="projecton",
+    M="margins",
+    R="region",
+    SC="sharex",
+    SR="sharey",
+    T="title",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+)
+@kwargs_to_strings(Ff="sequence", Fs="sequence", M="sequence", R="sequence")
+def subplot(self, nrows=1, ncols=1, **kwargs):
+    r"""
+    Create multi-panel subplot figures.
+
+    This function is used to split the current figure into a rectangular layout
+    of subplots that each may contain a single self-contained figure. Begin by
+    defining the layout of the entire multi-panel illustration. Several
+    parameters are available to specify the systematic layout, labeling,
+    dimensions, and more for the subplots.
+
+    Full option list at :gmt-docs:`subplot.html#synopsis-begin-mode`
+
+    {aliases}
+
+    Parameters
+    ----------
+    nrows : int
+        Number of vertical rows of the subplot grid.
+    ncols : int
+        Number of horizontal columns of the subplot grid.
+    figsize : tuple
+        Specify the final figure dimensions as (*width*, *height*).
+    subsize : tuple
+        Specify the dimensions of each subplot directly as (*width*, *height*).
+        Note that only one of ``figsize`` or ``subsize`` can be provided at
+        once.
+
+    autolabel : bool or str
+        [*autolabel*][**+c**\ *dx*\ [/*dy*]][**+g**\ *fill*][**+j**\|\ **J**\
+        *refpoint*][**+o**\ *dx*\ [/*dy*]][**+p**\ *pen*][**+r**\|\ **R**]
+        [**+v**].
+        Specify automatic tagging of each subplot. Append either a number or
+        letter [a]. This sets the tag of the first, top-left subplot and others
+        follow sequentially. Surround the number or letter by parentheses on
+        any side if these should be typeset as part of the tag. Use **+j|J**\
+        *refpoint* to specify where the tag should be placed in the subplot
+        [TL]. Note: **+j** sets the justification of the tag to *refpoint*
```

Comment:
When I review the PR #881 from @meghanrjones, I realize that the default values are not documented well. Most readers may find `[TL]` weird, and don't understand what it is, especially that `[ ]` means optional arguments in GMT's syntax. I feel that we should use `[Default is TL]`, rather than just `[TL]` for better clarification. 

If you agree, then you need to update the docstrings again, and better to document it in the contributing guides.


```suggestion
        [Default is **TL**]. Note: **+j** sets the justification of the tag to *refpoint*
```

Reply:
> When I review the PR #881 from @meghanrjones, I realize that the default values are not documented well. Most readers may find `[TL]` confusing, especially that `[ ]` means optional arguments in GMT's syntax. I feel that we should use `[Default is TL]`, rather than just `[TL]` for better clarification.

Yes, I hear you that using `[Default is TL]` is clearer, but:

1. There are a lot of these `[ ]`, and it's not easy to CTRL+F and replace all instances.
2. We have mostly been copying and pasting from upstream GMT, and it will be a lot of work checking through them all (c.f. #895).

> If you agree, then you need to update the docstrings again, and better to document it in the contributing guides.

I think it's ok for `[ ]` to mean both optional and default as long as it's documented, or like you say, use `[Default is TL]` instead. I would rather we settle on the standard we want before applying it here in this PR.

Reply:
OK, we can discuss it later.